### PR TITLE
Fixes and improvments for model HTML help and editor

### PR DESCRIPTION
**Pull request by pathmapper**
## Description

This is an attempt to fix some issues with the current model HTML help and the corresponding help editor and to improve the HTML help a bit, in detail:

- Fixes an unreported bug which added _Outputs_ always also to the _Input parameters_ section of the HTML help.
- Removes the header _Algorithm description_ to match the model HTML help with the help of native QGIS algorithms where such a header isn't used.
- Add only parameters with description and headers with parameters to fix the behavior that for models with a long list of parameters without description, the model HTML help gets bloated with all parameter names which are visible from the model interface anyway. With this PR it's now possible to create a help which looks exactly like the help of native QGIS models, e.g. adding only a short description text and nothing else.
- If available, include missing information like _Algorithm author_, _Help author_ and _Algorithm version_ to the HTML help preview of the help editor and use the same wording for them to match the appearance of the model HTML help.
- Don't include HTML tags for translations. 

**Model HTML help _before_ this PR:**
image

**Model HTML help after this PR:**
image

### CODE REVIEW

File: src/core/processing/qgsprocessingutils.cpp [hunk 1]
```diff
@@ -1068,15 +1068,15 @@ QString QgsProcessingUtils::formatHelpMapAsHtml( const QVariantMap &map, const Q
     }
   }
   if ( !outputs.isEmpty() )
-    s += QObject::tr( "<h2>Outputs</h2>\n" ) + outputs;
+    s += QStringLiteral( "<h2>" ) + QObject::tr( "Outputs" ) + QStringLiteral( "</h2>\n" ) + outputs;
 
   s += QLatin1String( "<br>" );
   if ( !map.value( QStringLiteral( "ALG_CREATOR" ) ).toString().isEmpty() )
-    s += QObject::tr( "<p align=\"right\">Algorithm author: %1</p>" ).arg( getText( QStringLiteral( "ALG_CREATOR" ) ) );
+    s += QStringLiteral( "<p align=\"right\">" ) + QObject::tr( "Algorithm author") + QStringLiteral( ": " ) + getText( QStringLiteral( "ALG_CREATOR" ) ) + QStringLiteral( "</p>" );
```
Comment (nyalldawson) [inline src/core/processing/qgsprocessingutils.cpp]: I'm personally never sure about this one -- don't we also need to translate ":" for languages which don't use a ":" character? @rduivenvoorde any insights?

**Comment by nyalldawson** [comment]
Great changes, thanks @pathmapper ! I've just the one outstanding question about whether ":" characters SHOULD be translated...

File: src/core/processing/qgsprocessingutils.cpp (hunk 1 above)
Comment (nirvn) [reply src/core/processing/qgsprocessingutils.cpp]: Yes we do. For eg in Khmer, : is ៖

Comment (nyalldawson) [reply src/core/processing/qgsprocessingutils.cpp]: Thanks @nirvn --  that clears up one long-standing mystery for me!

Comment (pathmapper) [reply src/core/processing/qgsprocessingutils.cpp]: Thank you for the clarification and example @nirvn, done in https://github.com/qgis/QGIS/pull/44935/commits/4a6c2508f8d26876f181e817159c51c889020fe3.

**Comment by pathmapper** [comment]
Thanks for the review @nyalldawson, much appreciated !

File: src/core/processing/qgsprocessingutils.cpp (hunk 1 above)
Comment (rduivenvoorde) [reply src/core/processing/qgsprocessingutils.cpp]: > Yes we do. For eg in Khmer, : is ៖

Ghee @nirvn, was'nt aware of that. Thanks!

**Review by nyalldawson — APPROVED** [review]

**Pull request merged**